### PR TITLE
Update bundle for new templating/twig features

### PR DESCRIPTION
**Pull request by WouterJ**
This integrates https://github.com/KnpLabs/KnpMenu/pull/189, it has to be merged after that PR was merged.

### CODE REVIEW

File: Templating/Helper/MenuHelper.php~ [hunk 1]
```diff
@@ -0,0 +1,70 @@
+<?php
```
Comment (WouterJ) [inline Templating/Helper/MenuHelper.php~]: Done (I keep forgetting creating a global gitignore on my new PC)

**Comment by stof** [comment]
Please don't commit the backup file of your editor

File: Resources/config/menu.xml [hunk 2]
```diff
@@ -17,6 +17,7 @@
         <parameter key="knp_menu.renderer.list.options" type="collection"></parameter>
         <parameter key="knp_menu.listener.voters.class">Knp\Bundle\MenuBundle\EventListener\VoterInitializerListener</parameter>
         <parameter key="knp_menu.voter.router.class">Knp\Menu\Matcher\Voter\RouteVoter</parameter>
+        <parameter key="knp_menu.manipulator.class">Knp\Menu\Util\MenuManipulator</parameter>
```
Comment (stof) [inline Resources/config/menu.xml]: please don't introduce new class parameters. Using them is discouraged in Symfony

**Comment by stof** [comment]
This also needs to be updated for changes done in the KnpMenu PR

**Comment by WouterJ** [comment]
Updated this PR too. The failures in this PR are caused because the PR on KnpMenu is not merged yet.

**Comment by WouterJ** [comment]
Can anyone rerun the tests to see if anything is missing (it should work now, as KnpLabs/KnpMenu#189 was merged)

**Comment by stof** [comment]
@WouterJ I relaunched the build

**Comment by stof** [comment]
And the build is still failing because the composer deps are still allowing the version of KnpMenu without these functions

**Comment by WouterJ** [comment]
@stof is the master branch of KnpMenu 2.1.0-dev or 2.0.x-dev? The composer alias says the first, but given that this is a new feature, I would expect it becomes 2.1.0.

**Comment by WouterJ** [comment]
ping @stof , we would really like to finish this KnpMenu 2.0 thing for the CmfMenuBundle. Can you please answer my question above?

**Comment by WouterJ** [comment]
Tests are finally passing. Ready to go imo

**Comment by WouterJ** [comment]
ping @stof

**Comment by lsmith77** [comment]
ping

File: composer.json [hunk 3]
```diff
@@ -19,7 +19,7 @@
         }
     ],
     "require":      {
-        "knplabs/knp-menu":         "~2.0",
+        "knplabs/knp-menu":         "~2.1@dev",
```
Comment (docteurklein) [inline composer.json]: should we keep `@dev` here?

**Comment by docteurklein** [comment]
LGTM, if you think it's OK i'll merge it, and tag

**Comment by stof** [comment]
@docteurklein I will work on the pending work on this bundle and the library in the next few days. I have a few things I would like to merge before releasing.

**Comment by docteurklein** [comment]
@stof that's good news :)  /cc @lsmith77

File: composer.json (hunk 3 above)
Comment (stof) [inline composer.json]: @WouterJ given that 2.1 has been released, please change this to ``~2.1`` only

File: Tests/Templating/MenuHelperTest.php [hunk 4]
```diff
@@ -29,41 +27,120 @@ public function testGet()
...
+            ->with($menu)
+            ->will($this->returnValue(false))
+        ;
+
+        $helper = new MenuHelper($this->getHelperMock(), $matcherMock, $this->getManipulatorMock());
+
+        $this->assertFalse($helper->isAncestor($menu));
+    }
+
+    protected function getHelperMock(array $methods = array())
```
Comment (stof) [inline Tests/Templating/MenuHelperTest.php]: please use private methods

File: Resources/config/menu.xml [hunk 5]
```diff
@@ -69,6 +70,8 @@
         <service id="knp_menu.voter.router" class="%knp_menu.voter.router.class%">
             <tag name="knp_menu.voter" request="true" />
         </service>
+
+        <service id="knp_menu.manipulator" class="Knp\Menu\Util\MenuManipulator" />
```
Comment (stof) [inline Resources/config/menu.xml]: this should be a private service IMO

**Comment by WouterJ** [comment]
Thanks @stof. I've updated the PR

**Comment by dbu** [comment]
looks like the php 7 build on travis just broke :-(

but i think we can merge it now. lets see what stof says.

**Comment by dbu** [comment]
i mean the php 7 setup travis is using, not anything we do in our code

**Comment by stof** [comment]
I restarted the PHP 7 job and it passed now. I suspect it was related to the AWS issues of today.

**Comment by dbu** [comment]
:+1:

**Pull request merged**